toar/toar_planetscope.ipynb

```diff
@@ -0,0 +1,474 @@
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Conversion coefficients: {1: 1.92992290459e-05, 2: 2.0401521894e-05, 3: 2.2723104298e-05, 4: 3.35095412863e-05}\n"
+     ]
+    }
+   ],
+   "source": [
+    "from xml.dom import minidom\n",
+    "\n",
+    "xmldoc = minidom.parse(\"20170623_180038_0f34_3B_AnalyticMS_metadata.xml\")\n",
+    "nodes = xmldoc.getElementsByTagName(\"ps:bandSpecificMetadata\")\n",
+    "\n",
+    "# XML parser refers to bands by numbers 1-4\n",
+    "coeffs = {}\n",
+    "for node in nodes:\n",
+    "    bn = node.getElementsByTagName(\"ps:bandNumber\")[0].firstChild.data\n",
+    "    if bn in ['1', '2', '3', '4']:\n",
+    "        i = int(bn)\n",
+    "        value = node.getElementsByTagName(\"ps:reflectanceCoefficient\")[0].firstChild.data\n",
+    "        coeffs[i] = float(value)\n",
+    "\n",
+    "print \"Conversion coefficients:\", coeffs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that the coefficients are all of order `1e-5`, and that the coefficient for NIR is significantly higher than the coefficient for blue. This is a big deal if your use case involves performing band math because a pixel with a `NIR/blue` ratio of `1.0` in the radiance image will have a `NIR/blue` ratio of `3.35/1.929=1.73` in the reflectance image!\n",
+    "\n",
+    "Most spectral indices are defined in terms of reflectance, not radiance."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 4: Convert Radiance to Reflectance"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Radiance is measured in SI units: W/m^2. Reflectance is a ratio from 0 to 1. The conversion is performed as a simple per-band scalar multiplication:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Red band radiance is from 0 to 21681\n",
+      "Red band reflectance is from 0.0 to 0.492659624285\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Multiply the Digital Number (DN) values in each band by the TOA reflectance coefficients\n",
+    "band_blue_reflectance = band_blue_radiance * coeffs[1]\n",
+    "band_green_reflectance = band_green_radiance * coeffs[2]\n",
+    "band_red_reflectance = band_red_radiance * coeffs[3]\n",
+    "band_nir_reflectance = band_nir_radiance * coeffs[4]\n",
+    "\n",
+    "import numpy as np\n",
+    "print \"Red band radiance is from {} to {}\".format(np.amin(band_red_radiance), np.amax(band_red_radiance))\n",
+    "print \"Red band reflectance is from {} to {}\".format(np.amin(band_red_reflectance), np.amax(band_red_reflectance))\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 5. Save the Radiance Image "
```

Comment:
I think this step is "Save the Reflectance Image". Not sure if you want to include this here, it is often the case that reflectance values are scaled by factor of 10000. This allows the image to be saved as an unsigned 16bit-Int, the same format as the native scaled radiance